src/ui/gtk/settings_cb.c

```diff
@@ -962,7 +970,7 @@ on_entry_dbg_property_pattern_activate(GtkEditable *unused_editable,
         0, -1));
 	g_strstrip(text);
 
-	if (0 != strcmp(text, old_pattern)) {
+	if (0 != strcmp(text, old_pattern) || text[0] == '\0') {
```

Comment:
It is best to say
`'\0' == text[0]`
That way, if you make a typo and forget one '=' you get an invalid assignment and therefore an error.
Plus it puts emphasis on the value being tested.